### ISSUE TITLE
Using bitname/wordpress-nginx results in "Could not connect to the database"

### ISSUE THREAD

**Issue by alexauch01** [bitnami/containers#81389]
### Name and Version

bitnami/worpress-nginx:latest

### What architecture are you using?

None

### What steps will reproduce the bug?

I'm currently using TRAEFIK as a reverse proxy that handles SSL certificates, which works fine with "bitnami/wordpress:latest"

However I want to switch towards the version of wordpress using NGINX (bitnami/wordpress:latest). However when I use this version I have 2 problems:

**1) No CSS is applied** 

**2) Only when I click on login I got a warning "The information you’re about to submit is not secure"  on http://mydomain.com/wp-login.php**

### What is the expected behavior?

1) CSS should be applied
2) Login should be done over HTTPS

### What do you see instead?

1) <img width="1662" alt="Image" src="https://github.com/user-attachments/assets/a6575057-7af7-40cb-ac14-fd2e706053f7" />

2)<img width="890" alt="Image" src="https://github.com/user-attachments/assets/bc3f9b70-6c65-4a6f-82b5-91ca57dd5da7" />

### Additional information

Here is my configuration:

**ENV**
```
# Traefik Variables
TRAEFIK_IMAGE_TAG=traefik:v3.2
TRAEFIK_LOG_LEVEL=DEBUG
# Passwords must be encoded using BCrypt https://hostingcanada.org/htpasswd-generator/
TRAEFIK_BASIC_AUTH=USER:PASSWORD
# WordPress Variables
WORDPRESS_MARIADB_IMAGE_TAG=bitnami/mariadb:latest
WORDPRESS_IMAGE_TAG= bitnami/wordpress-nginx:latest
WORDPRESS_DB_NAME=wordpressdb
WORDPRESS_DB_USER=NAME
WORDPRESS_DB_PASSWORD=PASSWORD
WORDPRESS_DB_ADMIN_PASSWORD=PASSWORD
WORDPRESS_TABLE_PREFIX=wp_
WORDPRESS_BLOG_NAME=BLOG_NAME
WORDPRESS_ADMIN_NAME=NAME
WORDPRESS_ADMIN_LASTNAME=NAMEX
WORDPRESS_ADMIN_USERNAME=NAMEY
WORDPRESS_ADMIN_PASSWORD=PASSWORD
WORDPRESS_ADMIN_EMAIL=name@mydomain.com
WORDPRESS_SMTP_ADDRESS=smtp.google.com
WORDPRESS_SMTP_PORT=587
WORDPRESS_SMTP_USER_NAME=NAME
WORDPRESS_SMTP_PASSWORD=PASSWORD
```

**docker-compose-traefik.yml**
```
networks:
  proxy:
name: proxy
    external: true
volumes:
  letsencrypt:
    name: letsencrypt
services:
  traefik:
    image: traefik:v3.2
    networks:
      - proxy
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - ./letsencrypt:/letsencrypt
      - ./config:/etc/traefik/config
      - ./logs:/var/log
      - ./traefik.yml:/etc/traefik/traefik.yml:ro
    ports:
      - "80:80"
      - "443:443"
      - "8080:8080"
    restart: unless-stopped`
```
**traefik.yml**
```
global:
  checkNewVersion: false
  sendAnonymousUsage: false

log:
  level: DEBUG
  format: common
  filepath: /var/log/traefik.log
accesslog:
  format: DEBUG
  filepath: /var/log/traefik-access.log

api:
  insecure: false
  dashboard: true

providers:
  docker:
    watch: true
    exposedByDefault: false
    endpoint: "unix:///var/run/docker.sock"
    network: proxy
  file:
    directory: /etc/traefik/config
    watch: true

entryPoints:
  web:
    address: :80
    http:
      redirections:
        entrypoint:
          to: websecure
          scheme: https
  websecure:
    address: :443
    http:
      middlewares:
        - no-www@file

certificatesResolvers:
  letsencrypt:
    acme:
      email: admin@mydomain.com
      storage: /letsencrypt/acme.json
      tlsChallenge: {}
```

**config/dynamic.yml**
```
http: 
  routers:
    api:
      rule: Host(`traefik.mydomain.com`) || Host(`www.traefik.mydomain.com`)
      middlewares: 
        - auth
      service: api@internal
      entryPoints: 
        - "websecure"
      tls:
        certResolver: "letsencrypt"

  middlewares:
    auth:
      basicAuth:
        users : 
          - "user:password"
    no-www:
      redirectRegex:
        regex: "^https?://www\\.(.*)"
        replacement: "https://${1}"
        permanent: true
```

**docker-compose-wp-nginx.yml**
```
networks:
 proxy:
  external: true

volumes:
  mariadb-data:
    name: mariadb-data
  wordpress_data:
    name: wordpress_data

services:
  mariadb:
    image: ${WORDPRESS_MARIADB_IMAGE_TAG}
    volumes:
      - mariadb-data:/var/lib/mysql
    environment:
      MARIADB_DATABASE: ${WORDPRESS_DB_NAME}
      MARIADB_USER: ${WORDPRESS_DB_USER}
      MARIADB_PASSWORD: ${WORDPRESS_DB_PASSWORD}
      MARIADB_ROOT_PASSWORD: ${WORDPRESS_DB_ADMIN_PASSWORD}
      MYSQL_TCP_PORT: 3306
    ports:
      - "3306:3306"
    networks:
      - proxy
      - 
  wordpress:
    image: ${WORDPRESS_IMAGE_TAG}
    volumes:
      - 'wordpress_data:/bitnami/wordpress'
    depends_on:
      - mariadb
    environment:
      WORDPRESS_DATABASE_HOST: mariadb
      WORDPRESS_DATABASE_PORT_NUMBER: 3306
      WORDPRESS_DATABASE_NAME: ${WORDPRESS_DB_NAME}
      WORDPRESS_DATABASE_USER: ${WORDPRESS_DB_USER}
      WORDPRESS_DATABASE_PASSWORD: ${WORDPRESS_DB_PASSWORD}
      WORDPRESS_TABLE_PREFIX: ${WORDPRESS_TABLE_PREFIX}
      WORDPRESS_BLOG_NAME: ${WORDPRESS_BLOG_NAME}
      WORDPRESS_FIRST_NAME: ${WORDPRESS_ADMIN_NAME}
      WORDPRESS_LAST_NAME: ${WORDPRESS_ADMIN_LASTNAME}
      WORDPRESS_USERNAME: ${WORDPRESS_ADMIN_USERNAME}
      WORDPRESS_PASSWORD: ${WORDPRESS_ADMIN_PASSWORD}
      WORDPRESS_EMAIL: ${WORDPRESS_ADMIN_EMAIL}
      WORDPRESS_SMTP_HOST: ${WORDPRESS_SMTP_ADDRESS}
      WORDPRESS_SMTP_PORT: ${WORDPRESS_SMTP_PORT}
      WORDPRESS_SMTP_USER: ${WORDPRESS_SMTP_USER_NAME}
      WORDPRESS_SMTP_PASSWORD: ${WORDPRESS_SMTP_PASSWORD}
    networks:
      - proxy
    labels:
      - traefik.enable=true
      - traefik.http.routers.wordpress-website1.tls=true
      - traefik.http.routers.wordpress-website1.tls.certresolver=letsencrypt
      - traefik.http.routers.wordpress-website1.rule=Host(`mydomain.com`) || Host(`www.mydomain.com`)
      - traefik.http.routers.wordpress-website1.entrypoints=websecure
      # - traefik.http.services.wordpress-website.loadbalancer.server.port=8080
    restart: unless-stopped`
```

** LOGS ON MARIADB CONTAINER **
_==> ** Starting MariaDB **
/opt/bitnami/mariadb/sbin/mysqld: Deprecated program name. It will be removed in a future release, use '/opt/bitnami/mariadb/sbin/mariadbd' instead
2025-05-19 17:02:10 0 [Note] Starting MariaDB 11.4.6-MariaDB source revision 63a5552e76feb60ff62725e0274268e35e7a77e8 server_uid G/1qRu7lnmZmA4boN0yCWS9Nj8M= as process 1
2025-05-19 17:02:10 0 [Note] InnoDB: Compressed tables use zlib 1.2.13
2025-05-19 17:02:10 0 [Note] InnoDB: Number of transaction pools: 1
2025-05-19 17:02:10 0 [Note] InnoDB: Using crc32 + pclmulqdq instructions
2025-05-19 17:02:10 0 [Note] mysqld: O_TMPFILE is not supported on /opt/bitnami/mariadb/tmp (disabling future attempts)
2025-05-19 17:02:10 0 [Note] InnoDB: Using Linux native AIO
2025-05-19 17:02:10 0 [Note] InnoDB: innodb_buffer_pool_size_max=128m, innodb_buffer_pool_size=128m
2025-05-19 17:02:10 0 [Note] InnoDB: Completed initialization of buffer pool
2025-05-19 17:02:10 0 [Note] InnoDB: Buffered log writes (block size=512 bytes)
2025-05-19 17:02:11 0 [Note] InnoDB: End of log at LSN=1907813
2025-05-19 17:02:11 0 [Note] InnoDB: Opened 3 undo tablespaces
2025-05-19 17:02:11 0 [Note] InnoDB: 128 rollback segments in 3 undo tablespaces are active.
2025-05-19 17:02:11 0 [Note] InnoDB: Setting file './ibtmp1' size to 12.000MiB. Physically writing the file full; Please wait ...
2025-05-19 17:02:11 0 [Note] InnoDB: File './ibtmp1' size is now 12.000MiB.
2025-05-19 17:02:11 0 [Note] InnoDB: log sequence number 1907813; transaction id 875
2025-05-19 17:02:11 0 [Note] InnoDB: Loading buffer pool(s) from /bitnami/mariadb/data/ib_buffer_pool
2025-05-19 17:02:11 0 [Note] Plugin 'FEEDBACK' is disabled.
2025-05-19 17:02:11 0 [Note] Plugin 'wsrep-provider' is disabled.
2025-05-19 17:02:11 0 [Note] InnoDB: Buffer pool(s) load completed at 250519 17:02:11
2025-05-19 17:02:12 0 [Note] Server socket created on IP: '0.0.0.0'.
2025-05-19 17:02:12 0 [Warning] 'proxies_priv' entry '@% root@5bd8b44c4612' ignored in --skip-name-resolve mode.
2025-05-19 17:02:12 0 [Note] mysqld: Event Scheduler: Loaded 0 events
2025-05-19 17:02:12 0 [Note] /opt/bitnami/mariadb/sbin/mysqld: ready for connections.
Version: '11.4.6-MariaDB'  socket: '/opt/bitnami/mariadb/tmp/mysql.sock'  port: 3306  Source distribution_

** LOGS OF WORDPRESS CONTAINER **
_Welcome to the Bitnami wordpress-nginx container
wordpress 17:02:03.52 INFO  ==> Subscribe to project updates by watching https://github.com/bitnami/containers
wordpress 17:02:03.53 INFO  ==> Did you know there are enterprise versions of the Bitnami catalog? For enhanced secure software supply chain features, unlimited pulls from Docker, LTS support, or application customization, see Bitnami Premium or Tanzu Application Catalog. See https://www.arrow.com/globalecs/na/vendors/bitnami/ for more information.
wordpress 17:02:03.53 INFO  ==> 
wordpress 17:02:03.53 INFO  ==> ** Starting WordPress setup **
wordpress 17:02:03.56 INFO  ==> Validating settings in NGINX_* env vars
Certificate request self-signature ok
subject=CN = example.com
wordpress 17:02:08.45 INFO  ==> No custom scripts in /docker-entrypoint-initdb.d
wordpress 17:02:08.45 INFO  ==> Initializing NGINX
realpath: /bitnami/nginx/conf/vhosts: No such file or directory
wordpress 17:02:08.51 INFO  ==> Configuring PHP options
wordpress 17:02:08.52 INFO  ==> Setting PHP expose_php option
wordpress 17:02:08.53 INFO  ==> Setting PHP output_buffering option
wordpress 17:02:08.55 INFO  ==> Validating settings in MYSQL_CLIENT_* env vars
wordpress 17:02:08.76 INFO  ==> Restoring persisted WordPress installation
wordpress 17:02:08.84 INFO  ==> Trying to connect to the database server
wordpress 17:02:15.47 INFO  ==> ** WordPress setup finished! **

wordpress 17:02:15.48 INFO  ==> Starting PHP-FPM...
wordpress 17:02:15.48 INFO  ==> Starting NGINX...
wordpress 17:02:15.50 INFO  ==> ** Starting PHP-FPM **
wordpress 17:02:15.50 INFO  ==> ** Starting NGINX **
172.18.0.2 - - [19/May/2025:17:02:29 +0000] "GET /wp-admin/ HTTP/1.1" 302 5 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36" "MYPUBLICIP"
172.18.0.2 - - [19/May/2025:17:02:29 +0000] "GET /wp-login.php?redirect_to=http%3A%2F%2Fmydomain.com%2Fwp-admin%2F&reauth=1 HTTP/1.1" 200 1935 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36" "MYPUBLICIP"
172.18.0.2 - - [19/May/2025:17:03:08 +0000] "POST /wp-admin/admin-ajax.php HTTP/1.1" 200 59 "https://mydomain.com/wp-admin/profile.php" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36" "MYPUBLICIP"
172.18.0.2 - - [19/May/2025:17:03:08 +0000] "GET /wp-includes/images/spinner-2x.gif HTTP/1.1" 200 7536 "https://mydomain.com/wp-admin/load-styles.php?c=0&dir=ltr&load%5Bchunk_0%5D=dashicons,admin-bar,common,forms,admin-menu,dashboard,list-tables,edit,revisions,media,themes,about,nav-menus,wp-pointer,widgets&load%5Bchunk_1%5D=,site-icon,l10n,buttons,wp-auth-check&ver=6.8.1" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36" "MYPUBLICIP"
172.18.0.2 - - [19/May/2025:17:03:08 +0000] "GET /wp-login.php?interim-login=1&wp_lang=en_US HTTP/1.1" 200 2074 "https://mydomain.com/wp-admin/profile.php" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36" "MYPUBLICIP"
172.18.0.2 - - [19/May/2025:17:05:00 +0000] "POST /wp-admin/admin-ajax.php HTTP/1.1" 200 59 "https://mydomain.com/wp-admin/profile.php" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36" "MYPUBLICIP"
172.18.0.2 - - [19/May/2025:17:05:09 +0000] "GET /wp-admin/profile.php HTTP/1.1" 302 5 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36" "1MYPUBLICIP"
172.18.0.2 - - [19/May/2025:17:05:09 +0000] "GET /wp-login.php?redirect_to=http%3A%2F%2Fmydomain.com%2Fwp-admin%2Fprofile.php&reauth=1 HTTP/1.1" 200 1937 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36" "MYPUBLICIP"
172.18.0.2 - - [19/May/2025:17:05:15 +0000] "GET /wp-login.php?redirect_to=http%3A%2F%2Fmydomain.com%2Fwp-admin%2Fprofile.php&reauth=1 HTTP/1.1" 200 1937 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36" "MYPUBLICIP"
2025/05/19 17:12:22 [error] 217#217: *12 access forbidden by rule, client: 172.18.0.2, server: _, request: "GET /.git/config HTTP/1.1", host: "mydomain.com"
172.18.0.2 - - [19/May/2025:17:12:22 +0000] "GET /.git/config HTTP/1.1" 403 179 "-" "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/74.0.3729.157 Safari/537.36" "196.251.88.164"_

**Comment by carrodher** [comment]
Hi, the issue may not be directly related to the Bitnami container image/Helm chart, but rather to how the application is being utilized, configured in your specific environment, or tied to a particular scenario that is not easy to reproduce on our side.

If you think that's not the case and would like to contribute a solution, we'd like to invite you to create a pull request. The Bitnami team is excited to review your submission and offer feedback. You can find the contributing guidelines [here](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md).

Your contribution will greatly benefit the community. Please don't hesitate to contact us if you have any questions or need help.

Suppose you have questions about the application, customizing its content, or using technology and infrastructure. In that case, we strongly recommend that you consult the forums and user guides provided by the project responsible for the application or technology.

With that said, we'll keep this ticket open until the stale bot automatically closes it, in case someone from the community contributes valuable insights.

**Comment by alexauch01** [comment]
I found a solution [here](https://community.traefik.io/t/traefik-and-nginx-wordpress/22115/7)

Adding this lines in wp-config.php fixed my issue:

```
    define('WP_SITEURL', 'https://domain.com');
    define('WP_HOME', 'https://domain.com');

    /** Fix for SSL behind Proxy **/
    $parsedUrl = parse_url(WP_HOME);
    $scheme = $parsedUrl['scheme'];
    if ($scheme == 'https') {$_SERVER['HTTPS'] = 'on';}
    else {$_SERVER['HTTPS'] = 'off';}
```
